mithril-common/src/signable_builder/cardano_immutable_full_signable_builder.rs

```diff
@@ -83,6 +84,14 @@ mod tests {
         ) -> Result<String, ImmutableDigesterError> {
             Ok(format!("immutable {}", beacon.immutable_file_number))
         }
+
+        async fn compute_merkle_tree(
+            &self,
+            _dirpath: &Path,
+            _beacon: &CardanoDbBeacon,
+        ) -> Result<MKTree<MKTreeStoreInMemory>, ImmutableDigesterError> {
+            unimplemented!()
```

Comment:
Just a suggestion to avoid the use of `unimplemented!` here, we could add the `process_immutables` function to the `ImmutableDigester` trait. This would allow to do a blanket implementation for both `compute_digest` and `compute_merkle_tree`.

Reply:
I don't think that's a good idea:
- `process_immutables` is an internal method, making it available in the public API would also force to make public internal types (`ComputedImmutablesDigests` in this case).
- The `ImmutableDigester` is used [on the public api of the `mithril-client` lib](https://docs.rs/mithril-client/latest/mithril_client/struct.MessageBuilder.html#method.with_immutable_digester) (to allow usage of other digesters by third party developers, as we do for other traits), so we should keep its API surface lean.
- `compute_merkle_tree` and `compute_digest` are not side effects free: they both use the logger and need to update the cache, two things only available on the structure that implement the `ImmutableDigester` trait.

Reply:
Approach to avoid `unimplemented!()` by using a pattern builder: https://github.com/input-output-hk/mithril/pull/2146/commits/ac1e197213d00f04b3a5f448e10bbe5075b55f7b

Reply:
I think on the opposite that there is no problem to even have the digester trait with a single function `compute_immutable_digests` that returns a `ComputedImmutablesDigests` which is responsible to compute both the digest and the Merkle tree. Apart form the fact that it would introduce a breaking change in the client, it would be even easier for third party developers to use our API.  The cache update should also probably be done by this function as computing the Merkle tree and the digest should be stateless operations.

OK with the builder approach to remove the usage of `unimplemented!` :+1: